app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,41 @@
+class PostsController < ApplicationController
+
+def index
+  @posts = Post.all
+end
+
+def create
+  @post = Post.create!(post_params)
+      redirect_to (post_path(@post))
+end
+
+def new
+    @post = Post.new
+  end
+
+  def show
+      @post = Post.find(params[:id])
+    end
+
+  def edit
+        @post = Post.find(params[:id])
+      end
+
+  def update
+    @post = Post.find(params[:id])
+    @post.update(artist_params)
+    redirect_to post_path(@post)
+  end
+
+  def destroy
+      @post = Post.find(params[:id])
+      @post.destroy
+      redirect_to post_path
```

Comment:
The post doesn't exist any more (yay successful delete action!), so trying to redirect to the `post_path` throws an error. What if you redirected to `posts_path` (the posts index page) instead?